test/e2e/delete_cm_test.go

```diff
@@ -0,0 +1,118 @@
+// +build deletecm
+
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	. "github.com/cloudevents/sdk-go/v2/test"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventing "knative.dev/eventing/pkg/apis/eventing/v1beta1"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/eventing/test/lib/recordevents"
+	"knative.dev/eventing/test/lib/resources"
+	"knative.dev/pkg/system"
+
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/kafka"
+)
+
+func TestDeleteCM(t *testing.T) {
+
+	const (
+		triggerName = "trigger"
+		brokerName  = "broker"
+		subscriber  = "subscriber"
+		senderName  = "sender"
+
+		configMapName = "kafka-broker-brokers-triggers"
+
+		eventType   = "type1"
+		eventSource = "source1"
+	)
+
+	eventId := uuid.New().String()
+
+	ctx := context.Background()
+
+	client := testlib.Setup(t, true)
+	defer testlib.TearDown(client)
+
+	client.CreateBrokerV1OrFail(
+		brokerName,
+		resources.WithBrokerClassForBrokerV1(kafka.BrokerClass),
+	)
+
+	eventTracker, _ := recordevents.StartEventRecordOrFail(ctx, client, subscriber)
+
+	client.CreateTriggerOrFailV1Beta1(
+		triggerName,
+		resources.WithBrokerV1Beta1(brokerName),
+		resources.WithSubscriberServiceRefForTriggerV1Beta1(subscriber),
+		func(trigger *eventing.Trigger) {
+			trigger.Spec.Filter = &eventing.TriggerFilter{
+				Attributes: map[string]string{
+					"source": eventSource,
+				},
+			}
+		},
+	)
+
+	client.WaitForAllTestResourcesReadyOrFail(ctx)
+
+	t.Logf("Sending events to %s/%s", client.Namespace, brokerName)
+
+	eventToSend := cloudevents.NewEvent()
+	eventToSend.SetID(eventId)
+	eventToSend.SetType(eventType)
+	eventToSend.SetSource(eventSource)
+
+	client.SendEventToAddressable(
+		ctx,
+		senderName+"matching",
+		brokerName,
+		testlib.BrokerTypeMeta,
+		eventToSend,
+	)
+
+	eventTracker.AssertAtLeast(1, recordevents.MatchEvent(HasId(eventId)))
+
+	t.Logf("Deleting ConfigMap %s", configMapName)
+
+	err := client.Kube.CoreV1().ConfigMaps(system.Namespace()).Delete(ctx, configMapName, metav1.DeleteOptions{})
+	assert.Nil(t, err)
+
+	client.WaitForAllTestResourcesReadyOrFail(ctx)
```

Comment:
Add 1/2 sec timeout here, just to make sure the controller reacts to the deleted CM

Reply:
Good idea, I added a poll loop (10s timeout) to reduce flakiness.

Reply:
Done.